department-of-veterans-affairs/abd-vro
# ABD-VRO/909: GitHub Actions: Send and attach "Failure" message to deploy workflow Slack message on any error

## What was the problem?
Attaches the "Deployment Failure" message to the "Deployment Start" thread in slack.

Associated tickets or Slack threads:

- https://github.com/department-of-veterans-affairs/abd-vro/issues/909
- 
## How does this fix it?[^1]
- we now output cancelled or success status from jobs that the slack notification job can consume

## How to test this PR
- Start a "Deploy to Dev" job
- - Wait for "Starting Deployment" message to go to slack channel
- Either 
-- 1. Cancel the job
-- 2. Cause an error (that may require writing something such as `exit 1` in the tasks - this is how I tested it)
- the "Deploy to Dev" thread should now either have 
-- 1. a "Deployment Cancelled" message if either of the jobs in the workflow were cancelled
-- 2. a "Deployment Failed!" message if either of the jobs in the workflow failed


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
